pymodbus/server/async_io.py

```diff
@@ -180,19 +180,23 @@ def execute(self, request, *addr):
         if self.server.request_tracer:
             self.server.request_tracer(request, *addr)
 
-        broadcast = False
+        if self.server.broadcast_enable and not request.slave_id:
+            # if broadcasting then execute on all slave contexts,
+            # note response will be ignored
+            for slave_id in self.server.context.slaves():
+                request.execute(self.server.context[slave_id])
+            # no response when broadcasting
+            return
+
+        slave_id = request.slave_id
         try:
-            if self.server.broadcast_enable and not request.slave_id:
-                broadcast = True
-                # if broadcasting then execute on all slave contexts,
-                # note response will be ignored
-                for slave_id in self.server.context.slaves():
-                    response = request.execute(self.server.context[slave_id])
-            else:
-                context = self.server.context[request.slave_id]
-                response = request.execute(context)
+            if self.server.is_server and self.comm_params.comm_type in (CommType.TCP,):
```

Comment:
Had the exact same question regarding is_server. It is [set to true by ModbusBaseServer](https://github.com/pymodbus-dev/pymodbus/blob/131d30391efd9c64f1fec857ae0372746374e6d2/pymodbus/server/async_io.py#L263C1-L266C10).
I only had time to elaborate TCP. Since UDP is also using a socket, it should be supported.
TLS I am not quite sure, because [the doc](https://docs.python.org/3/library/asyncio-protocol.html#asyncio.BaseTransport.get_extra_info) is not specific enough if the information is provided. I need to look into the tests to be sure.